prow/config/branch_protection.go

```diff
@@ -344,18 +344,22 @@ func (c *Config) GetBranchProtection(org, repo, branch string, presubmits []Pres
 		return nil, err
 	}
 
-	return c.GetPolicy(org, repo, branch, *b, presubmits)
+	return c.GetPolicy(org, repo, branch, *b, presubmits, nil)
 }
 
 // GetPolicy returns the protection policy for the branch, after merging in presubmits.
-func (c *Config) GetPolicy(org, repo, branch string, b Branch, presubmits []Presubmit) (*Policy, error) {
+func (c *Config) GetPolicy(org, repo, branch string, b Branch, presubmits []Presubmit, protectedOnGitHub *bool) (*Policy, error) {
 	policy := b.Policy
 
 	// Automatically require contexts from prow which must always be present
 	if prowContexts, requiredIfPresentContexts, optionalContexts := BranchRequirements(branch, presubmits); c.shouldManageRequiredStatusCheck(prowContexts, requiredIfPresentContexts, optionalContexts) {
 		// Error if protection is disabled
 		if policy.Protect != nil && !*policy.Protect {
 			if c.BranchProtection.AllowDisabledJobPolicies != nil && *c.BranchProtection.AllowDisabledJobPolicies {
+				if protectedOnGitHub != nil && *protectedOnGitHub {
+					logrus.WithField("branch", fmt.Sprintf("%s/%s/%s", org, repo, branch)).
+						Debug("Should use unmanaged instead")
```

Comment:
I'd be fine with bumping this to a warning, and I think we should provide a more detailed sentence. Something like "`protect: false` configuration causes branchprotector to not manage branch protection settings at all. If this a desired behavior, use `unmanaged: true` instead." (imagine if you simply see a random "Should use unmanaged instead" in the log flood. Instead of what?)